platforms/allwinner-d1/boards/lichee-rv/src/spim.rs

```diff
@@ -0,0 +1,224 @@
+    mnemos_alloc::containers::HeapArray,
+    registry::{uuid, Envelope, KernelHandle, Message, RegisteredDriver, ReplyTo, Uuid},
+    Kernel,
+};
+
+pub static SPI1_TX_DONE: WaitCell = WaitCell::new();
+
+pub struct Spim1 {
+    _x: (),
+}
+
+pub unsafe fn kernel_spim1(spi1: SPI_DBI, ccu: &mut CCU, gpio: &mut GPIO) -> Spim1 {
+    // Set clock rate (fixed to 2MHz), and enable the SPI peripheral
+    ccu.spi1_clk.write(|w| {
+        w.clk_gating().on(); // ?
+        w.clk_src_sel().hosc(); // base:  24 MHz
+        w.factor_n().n1(); // /1:    24 MHz
+        w.factor_m().variant(11); // /12:    2 MHz
+        w
+    });
+    ccu.spi_bgr.modify(|_r, w| {
+        w.spi1_gating().pass().spi1_rst().deassert();
+        w
+    });
+
+    // Map the pins
+    gpio.pd_cfg1.write(|w| {
+        w.pd10_select().spi1_cs_dbi_csx();
+        w.pd11_select().spi1_clk_dbi_sclk();
+        w.pd12_select().spi1_mosi_dbi_sdo();
+        w
+    });
+    gpio.pd_pull0.write(|w| {
+        w.pd10_pull().pull_disable();
+        w.pd11_pull().pull_disable();
+        w.pd12_pull().pull_disable();
+        w
+    });
+
+    // Hard coded configuration for specifically supporting the SHARP memory display
+
+    spi1.spi_gcr.write(|w| {
+        w.tp_en().normal();
+        w.mode().master();
+        w.en().enable();
+        w
+    });
+    spi1.spi_tcr.write(|w| {
+        w.ss_owner().spi_controller();
+        // w.cpol().low();
+        // w.cpha().p0();
+        w.fbs().lsb();
+        w.spol().clear_bit();
+        w
+    });
+    spi1.spi_fcr.modify(|_r, w| {
+        w.tf_drq_en().enable();
+        w
+    });
+
+    Spim1 { _x: () }
+}
+
+pub struct SpiSender {
+    _x: (),
+}
+
+impl SpiSender {
+    pub async fn register(kernel: &'static Kernel, queued: usize) -> Result<(), ()> {
+        let (kprod, kcons) = KChannel::new_async(kernel, queued).await.split();
+
+        kernel
+            .spawn(async move {
+                let kcons = kcons;
+                let spi = unsafe { &*SPI_DBI::PTR };
+
+                let txd_ptr: *mut u32 = spi.spi_txd.as_ptr();
+                let txd_ptr: *mut u8 = txd_ptr.cast();
+                let txd_ptr: *mut () = txd_ptr.cast();
+
+                loop {
+                    let msg: Message<SpiSender> = kcons.dequeue_async().await.unwrap();
+                    // println!("DEQUEUE");
+                    let Message { msg, reply } = msg;
+                    let SpiSenderRequest::Send(ref payload) = msg.body;
+
+                    spi.spi_bcc.modify(|_r, w| {
+                        w.stc().variant(payload.len() as u32);
+                        w
+                    });
+                    spi.spi_mbc.modify(|_r, w| {
+                        w.mbc().variant(payload.len() as u32);
+                        w
+                    });
+                    spi.spi_mtc.modify(|_r, w| {
+                        w.mwtc().variant(payload.len() as u32);
+                        w
+                    });
```

Comment:
it could be nice if there were comments here saying what these different registers do?